integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonDropDataMapCommand.scala

```diff
@@ -116,8 +116,12 @@ case class CarbonDropDataMapCommand(
         // drop index datamap on the main table
         if (mainTable != null &&
             DataMapStoreManager.getInstance().getAllDataMap(mainTable).size() > 0) {
-          dropDataMapFromSystemFolder(sparkSession)
-          return Seq.empty
+          val dmSchemaOp = DataMapStoreManager.getInstance().getAllDataMap(mainTable).asScala.
+            find(_.getDataMapSchema.getDataMapName.equalsIgnoreCase(dataMapName))
```

Comment:
I think you can use `.exists` instead of `.find` and move this check to line 118

Reply:
This if branch is for dropping index datamap and MV datamap, and can you also add comment for which if branch is for dropping preagg datamap